app/controllers/reports_controller.rb

```diff
@@ -0,0 +1,69 @@
+class ReportsController < ApplicationController
+  before_action :require_login
+  before_action :set_reports, only: %w(index show edit update destroy sort)
+  before_action :set_report, only: %w(show edit update destroy sort)
+  before_action :set_user, only: :show
+  respond_to :html, :json
+
+  def index
+  end
+
+  def show
+  end
+
+  def new
+    @report = Report.new
+    @report_flag = true
+    @report_date = Time.zone.today
+    @user_name = current_user.login_name
+  end
+
+  def edit
+    @report.user = current_user
+  end
+
+  def create
+    @report = Report.new(report_params)
+    @report.user = current_user
+    if @report.save
+      redirect_to @report, notice: t("report_was_successfully_created")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.